torch/distributions/transformed_distribution.py

```diff
@@ -55,7 +55,7 @@ def sample(self, sample_shape=torch.Size()):
         x = self.base_dist.sample(sample_shape)
         for transform in self.transforms:
             x = transform(x)
-        return x
+        return x.detach() if hasattr(x, 'detach') else x
```

Comment:
Is this equivalent to `isinstance(x, Variable)` but just future-proof to the Tensor/Variable merge?

Reply:
I think it's equivalent, but this version is more literally exactly what I want. It is copied from `Distribution.sample()`.